### PR TITLE
Update Golang version to 1.17 #173

### PR DESCRIPTION
**Pull request by seemiller**
Updated Go to 1.17

Updated go version references from 1.16 to 1.17 and ran:

```
go mod tidy -go=1.16 && go mod tidy -go=1.17
go mod vendor
hack/build-and-lint.sh
```

Goreleaser bug has been marked as closed and should be fixed in > 0.175.0. Updated goreleaser version to 0.184.0.

This is a new PR as the previous PR, #186, rotted with merge conflicts, plus there are some module versions changed as well.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,17 +1,58 @@
...
-go 1.16
+go 1.17
 
 require (
 	github.com/cppforlife/cobrautil v0.0.0-20200514214827-bb86e6965d72
 	github.com/cppforlife/go-cli-ui v0.0.0-20200716203538-1e47f820817f
 	github.com/google/go-containerregistry v0.7.0
 	github.com/hashicorp/go-version v1.3.0
-	github.com/kisielk/errcheck v1.6.0
+	github.com/kisielk/errcheck v1.5.0
```
Comment (joaopapereira) [inline go.mod]: Any idea why this got downgraded?

**Review by joaopapereira — COMMENTED** [review]
Added a question about one of the libraries that got downgraded.
Appart from that it looks good

File: go.mod (hunk 1 above)
Comment (seemiller) [reply go.mod]: Looks like the `v1.6.0` tag is associated to a release named `Add SelectorName` - https://github.com/kisielk/errcheck/releases

My guess is that there is a mixup with the release name not matching a version string.

Comment (seemiller) [reply go.mod]: I can update the PR to reference the `v1.6.0`tag

Comment (joaopapereira) [reply go.mod]: if you could I think it would be best

**Review by joaopapereira — APPROVED** [review]
Looks good, thanks

**Pull request merged**